core/src/main/scala/cats/data/Continuation.scala

```diff
@@ -0,0 +1,108 @@
+package cats
+package data
+
+import java.io.Serializable
+import scala.annotation.tailrec
+
+sealed abstract class Continuation[O, +I] extends Serializable {
```

Comment:
no, `O` is invariant. Note that it appears in the output, so it is either invariant or covariant. But it also appears in the input of the function, which makes it invariant or contravariant (see the `apply` method), so it is invariant.

A more direct way to see it: if you have a function `(I => O) => O` and say, `O = String`. Now due to covariance, I want to treat it as `(I => Any) => Any` can I do this? It is not clear at all how since internal to a function `(I => String) => String` we could make use of the String structure, but when I am given a function to Any, I can't do that.

So in fact, `Continuation[O, +I]` is the correct variance.

Now, about ordering of the parameters, I agree it is not the most intuitive order, but due to the the 2712 fix, right to left is the standard order searched when looking for single parameter type classes, so I think we are better off keeping the parameter there.